model/task_queue_service_dependency.go

```diff
@@ -321,6 +321,37 @@ func (d *basicCachedDAGDispatcherImpl) FindNextTask(ctx context.Context, spec Ta
 				continue
 			}
 
+			if err = nextTaskFromDB.SetGenerateTasksEstimations(ctx); err != nil {
```

Comment:
I think it makes sense for this also to be conditional if MaxPendingGeneratedTasks > 0

Reply:
I thought about this more and I think we also actually should set this field immediately upon task creation, and not here. This is because some of the scheduling limit work I'm doing requires that this field is set when we generate / create tasks to get an estimate of how many tasks are about to get scheduled.